ddp-workspace/projects/ddp-dsm-ui/src/app/tissue-list/tissue-list.component.ts

```diff
@@ -1660,12 +1665,59 @@ export class TissueListComponent implements OnInit {
   private getAssignees( realm: string ): void {
     this.dsmService.getAssignees(realm).subscribe({
       next: data => {
+        this.assignees = [];
         this.assignees.push( new Assignee( '-1', 'Remove Assignee', '' ) );
         data.forEach( ( val ) => {
           this.assignees.push( Assignee.parse( val ) );
         } );
+        let assigneesMap = this.getAssigneeAsNameValue();
+        if(!this.allColumns["data"].find( f => f.participantColumn.name === ParticipantColumn.ASSIGNEE_TISSUE.name))
+          this.allColumns["data"].push(new Filter( ParticipantColumn.ASSIGNEE_TISSUE, Filter.OPTION_TYPE, assigneesMap ));
+        for (let data of this.dataSources) {
+          this.allColumns[ data ].sort( ( a, b ) => {
+            return a.participantColumn.display.localeCompare( b.participantColumn.display );
+          } );
+        }
       },
       error: err => { console.log(err); }
     });
   }
+
+  getAssigneeById( assigneeId: any ) {
+    for (let assignee of this.assignees){
+      if(assignee.assigneeId === assigneeId)
+        return assignee.name;
+    }
+  }
+
+  getAssigneeAsNameValue(){
+    let assigneesMap = [];
+    if (this.assignees) {
+      this.assignees.forEach( assignee => {
+        if (assignee.assigneeId !== "-1") {
+          assigneesMap.push( new NameValue( assignee.assigneeId, assignee.name ) );
+        }
+      } );
+    }
+    return assigneesMap;
+  }
+
+  adjustAssigneeSavedFilterColumn(filter: Filter) {
+    let assigneesMap = this.getAssigneeAsNameValue();
+    let selectedOptions = filter.getSelectedOptionsBoolean( assigneesMap );
+    filter.selectedOptions = selectedOptions;
+    let f = this.selectedColumns[ 'data' ].find( filter => filter.participantColumn.name === ParticipantColumn.ASSIGNEE_TISSUE.name );
+    if (f) {
+      let index = this.selectedColumns[ 'data' ].indexOf( f );
+      this.selectedColumns[ 'data' ].splice( index, 1 );
+      f.selectedOptions = selectedOptions;
+      this.selectedColumns[ 'data' ].push( f );
+      for (let data of this.dataSources) {
```

Comment:
maybe we should rename that data to something else. makes it hard to read with the 'data' just a line up… maybe change it to source

Reply:
fixed